### ISSUE TITLE
current dev-hotfix/… branch causes "Contao core was about to get installed" loop

### ISSUE THREAD

**Issue by fritzmg** [contao-community-alliance/composer-client#278]
When updating the `composer-client` from `0.16.2` to `dev-hotfix/various-issues` it causes an infinite
```
[ContaoCommunityAlliance\Composer\Plugin\DuplicateContaoException]
Warning: Contao core was about to get installed but has been found in project root, to recover from this problem please restart the operation
```
loop, i.e. no matter how often you restart the `composer update` process after that, it will always fail with this message. The following is the complete log output after changing `"contao-community-alliance/composer-client": "~0.16"` to `"contao-community-alliance/composer-client": "dev-hotfix/various-issues"` in the `composer.json` and runing `composer update` 3 times:
```
# composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Installing contao-components/compass (0.12.2)
    Loading from cache

  - Installing true/punycode (1.1.0)
    Loading from cache

  - Installing tecnick.com/tcpdf (6.2.11)
    Loading from cache

  - Installing swiftmailer/swiftmailer (v5.4.1)
    Loading from cache

  - Installing simplepie/simplepie (1.3.1)
    Loading from cache

  - Installing phpspec/php-diff (v1.0.2)
    Loading from cache

  - Installing oyejorge/less.php (v1.7.0.10)
    Loading from cache

  - Installing michelf/php-markdown (1.6.0)
    Loading from cache

  - Installing leafo/scssphp (v0.6.3)
    Loading from cache



  [ContaoCommunityAlliance\Composer\Plugin\DuplicateContaoException]
  Warning: Contao core was about to get installed but has been found in project root, to recover from this problem please restart the operation


update [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-plugins] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--with-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--] [<packages>]...


# composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)


  [ContaoCommunityAlliance\Composer\Plugin\DuplicateContaoException]
  Warning: Contao core was about to get installed but has been found in project root, to recover from this problem please restart the operation


update [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-plugins] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--with-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--] [<packages>]...
```
Note that after changing the requirement to `dev-hotfix/various-issues` it also causes to download the `contao/core` dependencies again.

This issue also seems to be independent from the environment (tested under Windows 10 and Debian Wheezy, PHP 5.5 and 5.6).

When changing the requirement back to `~0.16` (for instance), the `composer update` runs normally and also removes the `contao/core` dependencies:
```
# composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Removing leafo/scssphp (v0.6.3)
  - Removing michelf/php-markdown (1.6.0)
  - Removing oyejorge/less.php (v1.7.0.10)
  - Removing phpspec/php-diff (v1.0.2)
  - Removing simplepie/simplepie (1.3.1)
  - Removing swiftmailer/swiftmailer (v5.4.1)
  - Removing tecnick.com/tcpdf (6.2.11)
  - Removing true/punycode (1.1.0)
  - Removing contao-components/compass (0.12.2)
Writing lock file
Generating autoload files
```

**Comment by fritzmg** [comment]
Doesn't happen with `0.16.3` anymore.

**Comment by discordier** [comment]
Problem was in plugin. Was fixed with 2.4.1. I just realized.

**Comment by fritzmg** [comment]
I see :)
